drivers/lsm6dsox/lsm6dsox.py

```diff
@@ -0,0 +1,228 @@
+"""
+LSM6DSOX STMicro driver for MicroPython based on LSM9DS1:
+Source repo: https://github.com/hoihu/projects/tree/master/raspi-hat
+
+The MIT License (MIT)
+
+Copyright (c) 2021 Damien P. George
+Copyright (c) 2021 Ibrahim Abdelkader <iabdalkader@openmv.io>
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+
+Basic example usage:
+
+import time
+from lsm6dsox import LSM6DSOX
+
+from machine import Pin, I2C
+lsm = LSM6DSOX(I2C(0, scl=Pin(13), sda=Pin(12)))
+
+while (True):
+    print('Accelerometer: x:{:>8.3f} y:{:>8.3f} z:{:>8.3f}'.format(*lsm.read_accel()))
+    print('Gyroscope:     x:{:>8.3f} y:{:>8.3f} z:{:>8.3f}'.format(*lsm.read_gyro()))
+    print("")
+    time.sleep_ms(100)
+"""
+import array
+
+
+class LSM6DSOX:
+    CTRL3_C = const(0x12)
+    CTRL1_XL = const(0x10)
+    CTRL8_XL = const(0x17)
+    CTRL9_XL = const(0x18)
+
+    CTRL2_G = const(0x11)
+    CTRL7_G = const(0x16)
+
+    OUTX_L_G = const(0x22)
+    OUTX_L_XL = const(0x28)
+    MLC_STATUS = const(0x38)
+
+    DEFAULT_ADDR = const(0x6A)
+    WHO_AM_I_REG = const(0x0F)
+
+    FUNC_CFG_ACCESS = const(0x01)
+    FUNC_CFG_BANK_USER = const(0)
+    FUNC_CFG_BANK_HUB = const(1)
+    FUNC_CFG_BANK_EMBED = const(2)
+
+    MLC0_SRC = const(0x70)
+    MLC_INT1 = const(0x0D)
+    TAP_CFG0 = const(0x56)
+
+    EMB_FUNC_EN_A = const(0x04)
+    EMB_FUNC_EN_B = const(0x05)
```

Comment:
Are these constants intended to be public (part of the API of this class) or private?  If they are private then it's best to prefix them with `_` because then they are fully optimised away and do not contribute at all to the bytecode.  That would save quite a bit of RAM (especially when compiled to .mpy).

Reply:
Private, prefixed with `_`